packages/workspace/src/browser/workspace-service.ts

```diff
@@ -120,9 +120,12 @@ export class WorkspaceService implements FrontendApplicationContribution {
             return undefined;
         }
         try {
-            const fileStat = await this.fileSystem.getFileStat(uri);
-            if (fileStat.isDirectory) {
-                return fileStat;
+            if (await this.fileSystem.exists(uri)) {
+                const fileStat = await this.fileSystem.getFileStat(uri);
```

Comment:
Should we consider to make `getFileStat` to return `undefined` instead of throwing?

cc @kittaakos @svenefftinge

Reply:
+1

Reply:
Good idea, I thought about it but didn't want to change the API for this PR, but if it's fine with everybody I can do that instead.

Reply:
@epatpol yes, it would be nice to explore this option, could you make a minimal PR?

Reply:
#1722